Pull request 8019 leads to compilation problems in step-62

When compiling step-62, I get the following errors:
```
~/git-files/dealii_anates/examples/step-62/step-62.cc: In member function ‘void step62::ElasticWave<dim>::assemble_system(double, bool)’:
~/git-files/dealii_anates/examples/step-62/step-62.cc:965:24: error: ‘i’ was not declared in this scope
               cell_rhs(i) += quadrature_data.right_hand_side[i];
                        ^
~/git-files/dealii_anates/examples/step-62/step-62.cc:965:24: note: suggested alternative: ‘xi’
               cell_rhs(i) += quadrature_data.right_hand_side[i];

```
and 
```
~/git-files/dealii_anates/examples/step-62/step-62.cc: In member function ‘virtual std::complex<double> step62::PML<dim>::value(const dealii::Point<dim>&, unsigned int) const’:
~/git-files/dealii_anates/examples/step-62/step-62.cc:480:14: error: no match for ‘operator+’ (operand types are ‘int’ and ‘std::complex<double>’)
     return 1 + std::max(calculated_pml_x_coeff, calculated_pml_y_coeff) *
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                  std::complex<double>(0, 1);
```
Based on what I can see, those changes were introduced in https://github.com/dealii/dealii/pull/8019. Is that a problem in my configuration, or a general bug (especially the first error)?

--- a/examples/step-62/step-62.cc
+++ b/examples/step-62/step-62.cc
@@ -477,8 +477,8 @@ namespace step62
           }
       }
 
-    return 1 + std::max(calculated_pml_x_coeff, calculated_pml_y_coeff) *
-                 std::complex<double>(0, 1);
+    return 1. + std::max(calculated_pml_x_coeff, calculated_pml_y_coeff) *
+                  std::complex<double>(0., 1.);
   }
 
 

